### PR TITLE
Introduce @Separator annotation for query parameters

### LINKED ISSUE #29528: Resteasy Reactive support for multivalued parameters
### Description

I've been trying to get Resteasy Reactive to parse arrays of values in my URLs in the form of comma separated values rather than '&' separated values.
e.g. **?array=item1,item2,item3** rather than **?array=item1&array=item2&array=item3**

This feature is available in the Resteasy classic extension through the usage of the @Separator annotation and the MultiValuedParamConverterProvider which comes included in resteasy-core and seems to be registered by default.

It would look something like:
```
@GET
public String hello(@QueryParam("foo") @Separator(",") List<String> foo) {}
```

It would be nice to have a similar way to do this for the Resteasy reactive extension. Possibly by including [these](https://github.com/resteasy/resteasy/tree/main/resteasy-core/src/main/java/org/jboss/resteasy/plugins/providers) providers which are used by default in the Resteasy classic extension.

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by geoand**
This annotation allows to actually turn a String into a List<String> automatically for each query param value

Resolves: #29528

P.S. A follow up that we might want to have is to introduce validation on the annotation's use

### CODE REVIEW

File: independent-projects/resteasy-reactive/common/runtime/src/main/java/org/jboss/resteasy/reactive/Separator.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import java.lang.annotation.Target;
+import java.util.List;
+
+/**
+ * When used on a {@link List} parameter annotated with {@link RestQuery}, RESTEasy Reactive will split the value of the
+ * parameter (using the value of the annotation) and populate the list with those values.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ ElementType.PARAMETER })
+public @interface Separator {
```
Comment (gsmet) [inline independent-projects/resteasy-reactive/common/runtime/src/main/java/org/jboss/resteasy/reactive/Separator.java:15]: Should it be included in the annotation mapping in the guide? IIRC, we have a table with the Classic -> Reactive mapping.

Comment (geoand) [reply independent-projects/resteasy-reactive/common/runtime/src/main/java/org/jboss/resteasy/reactive/Separator.java:15]: Very good point.

Comment (geoand) [reply independent-projects/resteasy-reactive/common/runtime/src/main/java/org/jboss/resteasy/reactive/Separator.java:15]: Done

**Comment by geoand** [comment]
@Sgitario what do you think about this one?

**Review by Sgitario — APPROVED** [review]
I can't say I'm a big fan of this feature, but if this is something that already exists in Resteasy Classic, go for it. 
The implementation looks really good to me!

**Comment by luca-bassoricci** [comment]
@geoand 
I see `@Separator` is valid only for parameters, but what about `@BeanParam` as in below example class?
```
public class BeanUsedAsBeanParam
{
	@RestQuery @Separator(",")
	Set<Long> longList;
	@RestQuery @Separator(",")
	Set<String> stringList;
}

```
Is that not possible at all or should be a valid use case?

**Comment by geoand** [comment]
It is a valid use case. We'll need to take care of that as well

**Comment by geoand** [comment]
https://github.com/quarkusio/quarkus/pull/29735 takes care of it

**Comment by geoand** [comment]
It seems we forgot to add it, so I opened https://github.com/quarkusio/quarkus/pull/41146 to include it.

**Comment by OndraZizka** [comment]
Hi, what do I need to do to make this work? It's not on classpath in the standard Quarkus.

**Comment by OndraZizka** [comment]
For those looking for this:
The FQCN is  `@org.jboss.resteasy.reactive.Separator` and it is in Quarkus when using `quarkus-rest`.

**Pull request merged**